src/autora/state/bundled.py

```diff
@@ -0,0 +1,175 @@
+
+        The `variables` can be updated using a `Delta`:
+        >>> dv1 = Delta(variables=VariableCollection(independent_variables=[Variable("1")]))
+        >>> s + dv1
+        StandardState(variables=VariableCollection(independent_variables=[Variable(name='1',...)
+
+        ... and are replaced by each `Delta`:
+        >>> dv2 = Delta(variables=VariableCollection(independent_variables=[Variable("2")]))
+        >>> s + dv1 + dv2
+        StandardState(variables=VariableCollection(independent_variables=[Variable(name='2',...)
+
+        The `conditions` can be updated using a `Delta`:
+        >>> dc1 = Delta(conditions=pd.DataFrame({"x": [1, 2, 3]}))
+        >>> (s + dc1).conditions
+           x
+        0  1
+        1  2
+        2  3
+
+        ... and are replaced by each `Delta`:
+        >>> dc2 = Delta(conditions=pd.DataFrame({"x": [4, 5]}))
+        >>> (s + dc1 + dc2).conditions
+           x
+        0  4
+        1  5
+
+        Datatypes other than `pd.DataFrame` will be coerced into a `DataFrame` if possible.
+        >>> import numpy as np
+        >>> dc3 = Delta(conditions=np.core.records.fromrecords([(8, "h"), (9, "i")], names="n,c"))
+        >>> (s + dc3).conditions
+           n  c
+        0  8  h
+        1  9  i
+
+        If they are passed without column names, no column names are inferred.
+        This is to ensure that accidental mislabeling of columns cannot occur.
+        Column names should usually be provided.
+        >>> dc4 = Delta(conditions=[(6,), (7,)])
+        >>> (s + dc4).conditions
+           0
+        0  6
+        1  7
+
+        Datatypes which are incompatible with a pd.DataFrame will throw an error:
+        >>> s + Delta(conditions="not compatible with pd.DataFrame")
+        Traceback (most recent call last):
+        ...
+        ValueError: ...
+
+        Experiment data can be updated using a Delta:
+        >>> ded1 = Delta(experiment_data=pd.DataFrame({"x": [1,2,3], "y": ["a", "b", "c"]}))
+        >>> (s + ded1).experiment_data
+           x  y
+        0  1  a
+        1  2  b
+        2  3  c
+
+        ... and are extended with each Delta:
+        >>> ded2 = Delta(experiment_data=pd.DataFrame({"x": [4, 5, 6], "y": ["d", "e", "f"]}))
+        >>> (s + ded1 + ded2).experiment_data
+           x  y
+        0  1  a
+        1  2  b
+        2  3  c
+        3  4  d
+        4  5  e
+        5  6  f
+
+        If they are passed without column names, no column names are inferred.
+        This is to ensure that accidental mislabeling of columns cannot occur.
+        >>> ded3 = Delta(experiment_data=pd.DataFrame([(7, "g"), (8, "h")]))
+        >>> (s + ded3).experiment_data
+           0  1
+        0  7  g
+        1  8  h
+
+        If there are already data present, the column names must match.
+        >>> (s + ded2 + ded3).experiment_data
+             x    y    0    1
+        0  4.0    d  NaN  NaN
```

Comment:
just curious, why do the values output with single decimal (as floats?) in this case? is there something about a NaN being introduced which triggers the switch? or are floats just the default when there are NaN contained in a data set?

Reply:
You're right, it's the float missing value which does it. Pandas uses np.nan (a float) internally (at least at the moment) so it promotes the integer to float values. They are thinking about how to use a different datatype for missing values in future.
https://pandas.pydata.org/docs/user_guide/missing_data.html#integer-dtypes-and-missing-data

Reply:
gotcha, thanks!